### ISSUE TITLE
Core dump in Aircrack-ng 1.0 Beta when loading capture file using -C option

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#346]
**Reported by jords on 17 Oct 2007 08:04 UTC**

I'm getting a core dump when loading a particular .cap file with aircrack-1.0 beta version. (I know, development version :D). I don't get a core dump when loading the .cap file if I don't use the new -C option to merge a list of MAC addresses.

This is the command I'm using:
```
# aircrack-ng  capture-01.cap -C 00:1B:0D:D5:47:80,00:1B:0D:D5:27:20,00:1C:0E:26:A2:C0,00:1B:0D:D5:42:40,00:1C:57:8A:B0:C0
And the output I get:
Opening capture-01.cap
*** glibc detected *** aircrack-ng: munmap_chunk(): invalid pointer: 0x08148674 ***
======= Backtrace: =========
/lib/tls/i686/cmov/libc.so.6(cfree+0x1bb)[Memory map: ========
08048000-0805a000 r-xp 00000000 08:02 1477625    /usr/local/bin/aircrack-ng
0805a000-0805c000 rw-p 00011000 08:02 1477625    /usr/local/bin/aircrack-ng
0805c000-08162000 rw-p 0805c000 00:00 0          [heap](0xb7d2892b]
aircrack-ng[0x804aa4c]
aircrack-ng[0x804afa8]
/lib/tls/i686/cmov/libpthread.so.0[0xb7f8f46b]
/lib/tls/i686/cmov/libc.so.6(clone+0x5e)[0xb7d8e73e]
=======)
b749f000-b74a0000 ---p b749f000 00:00 0
b74a0000-b7ca2000 rw-p b74a0000 00:00 0
b7ca2000-b7cb6000 r-xp 00000000 08:02 705985     /usr/lib/libz.so.1.2.3.3
b7cb6000-b7cb7000 rw-p 00013000 08:02 705985     /usr/lib/libz.so.1.2.3.3
b7cb7000-b7cb9000 r-xp 00000000 08:02 164328     /lib/tls/i686/cmov/libdl-2.6.1.so
b7cb9000-b7cbb000 rw-p 00001000 08:02 164328     /lib/tls/i686/cmov/libdl-2.6.1.so
b7cbb000-b7dff000 r-xp 00000000 08:02 164265     /lib/tls/i686/cmov/libc-2.6.1.so
b7dff000-b7e00000 r--p 00143000 08:02 164265     /lib/tls/i686/cmov/libc-2.6.1.so
b7e00000-b7e02000 rw-p 00144000 08:02 164265     /lib/tls/i686/cmov/libc-2.6.1.so
b7e02000-b7e05000 rw-p b7e02000 00:00 0
b7e05000-b7f30000 r-xp 00000000 08:02 246243     /usr/lib/i686/cmov/libcrypto.so.0.9.8
b7f30000-b7f45000 rw-p 0012a000 08:02 246243     /usr/lib/i686/cmov/libcrypto.so.0.9.8
b7f45000-b7f49000 rw-p b7f45000 00:00 0
b7f49000-b7f86000 r-xp 00000000 08:02 246244     /usr/lib/i686/cmov/libssl.so.0.9.8
b7f86000-b7f8a000 rw-p 0003c000 08:02 246244     /usr/lib/i686/cmov/libssl.so.0.9.8
b7f8a000-b7f9e000 r-xp 00000000 08:02 164339     /lib/tls/i686/cmov/libpthread-2.6.1.so
b7f9e000-b7fa0000 rw-p 00013000 08:02 164339     /lib/tls/i686/cmov/libpthread-2.6.1.so
b7fa0000-b7fa2000 rw-p b7fa0000 00:00 0
b7fab000-b7fb5000 r-xp 00000000 08:02 131349     /lib/libgcc_s.so.1
b7fb5000-b7fb6000 rw-p 0000a000 08:02 131349     /lib/libgcc_s.so.1
b7fb6000-b7fb9000 rw-p b7fb6000 00:00 0
b7fb9000-b7fd3000 r-xp 00000000 08:02 131542     /lib/ld-2.6.1.so
b7fd3000-b7fd5000 rw-p 00019000 08:02 131542     /lib/ld-2.6.1.so
bfb7c000-bfb92000 rw-p bfb7c000 00:00 0          [r-xp 00000000 00:00 0          [vdso](stack]
ffffe000-fffff000)
Aborted (core dumped)
```

This is completely reproducible; happens every time, unless like I said I leave off the -C option.
I'll attach the .cap file i'm trying to crack.

**Comment by aircrack-ng** [comment]
**Comment by jords on 17 Oct 2007 08:04 UTC**

Ok, decided to upload the .cap to a file upload site instead since even bzipped it's 19mb.

Here it is:
http: //rapidshare.com/files/63144184/capture-01.cap.bz2

**Comment by aircrack-ng** [comment]
**Comment by misterx on 17 Oct 2007 08:04 UTC**

cosmetic change

**Comment by aircrack-ng** [comment]
**Comment by hirte on 17 Oct 2007 08:04 UTC**

(In #797) Fixed aircrack-ng -C mergebssids bug (Closes: #346).
